### ISSUE TITLE
Potential bug in 3d bbox computation

### ISSUE THREAD

**Issue by FabianIsensee** [JunMa11/CVPR-MedSegFMCompetition#7]
When computing the middle slice in mask3D_to_bbox the middle slice is selected as `z_middle = z_indices[len(z_indices)//2]` where `z_indices, y_indices, x_indices = np.where(gt3D > 0)`. This will not yield the middle slice but the slice where the median voxel as found by np.where is located, which is potentially in conflict with the user instructions as described in [here](https://github.com/JunMa11/CVPR-MedSegFMCompetition/issues/5#issuecomment-2802778808). This particularly skews the results when the object size changes a lot between slices. To my eyes the correct implementation should be 

```
            unique_z = np.unique(z_indices)
            z_middle = unique_z[len(unique_z)//2]
```

Or is there a specific reason for implementing it the way you did?
Best,
Fabian

**Comment by kmin940** [comment]
Hi @FabianIsensee,

Thank you for reviewing our script for bounding box generation and you are absolutely correct. We will fix the bounding box and update the validation set. We appreciate your support!

**Comment by kmin940** [comment]
We have updated the code and the validation dataset on huggingface!

**Comment by kmin940** [comment]
We have updated the code and the validation dataset on huggingface!

**Comment by yuyi1005** [comment]
I optimized my training code to make it works better for bbox from the median voxel (consistent with validation), and now the validation code is updated? Then some of my efforts have been in vain! I don't think the validation code should be change at this point.
